prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -641,30 +641,48 @@ def prompt_processor() -> Any:
     return structured_output
 
 
-def simple_retriver(  # type:ignore
+def simple_or_subq_retriever(  # type:ignore
```

Comment:
@harini-venkataraman Let's rename this fn to `run_retrieval()`.

Reply:
@hari-kuriakose I had the same dilemma, of making the function name generic to all retrievals or keeping it specific. When other retrieval methods are introduced, felt this naming will be explanatory. Your thoughts?

Reply:
@harini-venkataraman The function name would still be `run_retrieval()`.
And we would add more retrieval strategies to the same if else ladder.

If the logic for a particular retrieval strategy becomes complex, then we could branch out that alone to a private member function, that's all. Th entrypoint for all retrieval strategies would continue to be `run_retrieval()`.